resolwe_bio/processes/workflows/quantseq.py

```diff
@@ -0,0 +1,369 @@
+        )
+        downsampling = GroupField(
+            Downsampling,
+            label="Downsampling with Seqtk",
+        )
+        preprocessing = GroupField(
+            Preprocessing,
+            label="Preprocessing with BBDuk",
+            hidden="trimming_tool != 'bbduk'",
+        )
+
+    class Output:
+        """Output fields."""
+
+    def run(self, inputs, outputs):
+        """Run the workflow."""
+
+        if inputs.trimming_tool == "bbduk" and not inputs.adapters:
+            self.error(
+                "Please provide fasta file of adapters, if you want to use BBDuk as a trimming tool."
+            )
+
+        if inputs.adapters:
+
+            input_preprocessing = {
+                "reads": inputs.reads,
+                "min_length": 20,
+                "reference": {"sequences": inputs.adapters},
+                "processing": {"kmer_length": 13},
+                "operations": {
+                    "k_trim": "r",
+                    "min_k": 6,
+                    "quality_trim": "r",
+                    "trim_quality": 10,
+                    "quality_encoding_offset": inputs.preprocessing.quality_encoding_offset,
+                    "ignore_bad_quality": inputs.preprocessing.ignore_bad_quality,
+                },
+                "fastqc": {
+                    "nogroup": True,
+                },
+            }
+
+            if inputs.reads.type.startswith("data:reads:fastq:single:"):
+                process_slug = "bbduk-single"
+            elif inputs.reads.type.startswith("data:reads:fastq:paired:"):
+                process_slug = "bbduk-paired"
+            else:
+                self.error("Wrong reads input type was provided.")
+
+        else:
+            if inputs.reads.type.startswith("data:reads:fastq:single:"):
+                input_preprocessing = {
+                    "reads": inputs.reads,
+                    "options": {
+                        "quality_cutoff": inputs.cutadapt.quality_cutoff,
+                    },
+                }
+                process_slug = "cutadapt-3prime-single"
+            else:
+                self.error("Wrong reads input type was provided.")
+
+        preprocessing = Data.create(
+            process=BioProcess.get_latest(process_slug), input=input_preprocessing
+        )
+
+        input_star = {
+            "reads": preprocessing,
+            "genome": inputs.genome,
+            "filtering": {
+                "out_filter_type": "BySJout",
+                "out_multimap_max": 20,
+                "out_mismatch_max": 999,
+                "out_mismatch_nl_max": 0.6,
+            },
+            "alignment": {
+                "align_overhang_min": 8,
+                "align_sjdb_overhang_min": 1,
+                "align_intron_size_min": 20,
+                "align_intron_size_max": 1000000,
+                "align_gap_max": 1000000,
+            },
+        }
+
+        if not inputs.adapters:
```

Comment:
It is possible for a user to provide an adapters file without selecting the bbduk as a trimming tool (e.g. ReSDK). It is much better to use checks for `inputs.trimming_tool` when preparing workflow inputs.